Automattic/jetpack
# Publicize: Reinstate the connect an account link

This changes ensures there's a link to connect an account at all times in the 'Share this post' panel in the sidebar.





Fixes #33172

## Proposed changes:

With no connections, we now show the 'Connect an account' link again.

<img width="285" alt="image" src="https://github.com/Automattic/jetpack/assets/96462/168deaad-1a1c-4026-83ec-6074871be65d">

This also refactors the connection refresh logic, so it's not tied to the PublicizeSettingsButton.

### Other information:


## Does this pull request change what data or activity we track or use?
No

## Testing instructions:
* On a fresh site, enable auto-sharing or install Jetpack Social
* Open the editor on a new post
* Check that you are shown the Instagram notice - dismiss it.
* It should be replaced by a 'Connect an account' link
* Click it and connect an account
* Return to the page and check the UI updates after 2 seconds
